dbms/src/Parsers/ParserInsertQuery.cpp

```diff
@@ -49,8 +51,10 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     if (!s_insert_into.ignore(pos, expected))
         return false;
 
-    if (table_function_p.parse(pos, table_function, expected))
+    if (s_function.ignore(pos, expected) || (s_table.ignore(pos, expected) && s_function.ignore(pos, expected)))
```

Comment:
If we introduced syntax `INSERT INTO FUNCTION f()`, `INSERT INTO TABLE FUNCTION f()`, we must also add support for `INSERT INTO TABLE t` in addition to `INSERT INTO t` for consistency.